packages/jest-transform/src/ScriptTransformer.ts

```diff
@@ -131,12 +133,18 @@ export default class ScriptTransformer {
   }
 
   private _getTransformPath(filename: Config.Path) {
-    for (let i = 0; i < this._config.transform.length; i++) {
-      if (new RegExp(this._config.transform[i][0]).test(filename)) {
-        return this._config.transform[i][1];
+    const transformRegExp = this._cache.transformRegExp;
+    if (!transformRegExp) {
+      return undefined;
+    }
+
+    for (const [regExp, transform] of transformRegExp) {
```

Comment:
Is for of faster than regular for loop now? I was under impression that using iterators and array destructuring (do we transpile it?) is still slower

Reply:
benchmark of this code running:
```
  for of x 580 ops/sec ±1.87% (93 runs sampled)
  for i  x 667 ops/sec ±1.91% (94 runs sampled)
```

using `for i` is more performant. i did it this way just for readability.

it's a hot code path so i should probably change it, thanks for calling that out